Mathlib/GroupTheory/QuotientGroup/Defs.lean

```diff
@@ -216,6 +225,23 @@ theorem map_comp_map {I : Type*} [Group I] (M : Subgroup H) (O : Subgroup I) [M.
     (map M O g hg).comp (map N M f hf) = map N O (g.comp f) hgf :=
   MonoidHom.ext (map_map N M O f g hf hg hgf)
 
+@[to_additive (attr := simp)]
+theorem comap_map_mk' (N H : Subgroup G) [N.Normal] :
+    Subgroup.comap (mk' N) (Subgroup.map (mk' N) H) = N ⊔ H :=
+  by simp [Subgroup.comap_map_eq, sup_comm]
+
+/-- The **correspondence theorem**, or lattice theorem,
```

Comment:
We try to keep the `Defs.lean` files somewhat minimal. Although indeed this result doesn't need any extra imports, I believe it's a better fit, together with the other three isomorphism theorems, in `QuotientGroup/Basic.lean`.

Reply:
The following suggestion is held with a weaker belief: `le_comap_mk'` and `comap_map_mk'` can probably also go to `Basic.lean`. I would keep `range_mk'` in `Defs.lean` since to me this is no different from `mk'_surjective`, which we also have here.

Reply:
`range_mk'` is retained in `Defs`, whereas `le_comap_mk'` and `comap_map_mk'` are put in a section for the development of the correspondence theorem in `Basics`. Ideally `Basics` may have a separate section for standard results with some standard development shared with other algebraic structures; and these results go there, but that's out of scope.